common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,590 @@
+        return gearOpt.isEmpty() ? false : gearOpt.get().getGearType() == gearItemReference.getGearType();
     }
 
     private void onSelectKeyPress(Slot hoveredSlot) {
         if (hoveredSlot == null) return;
+        // Selecting items behaves weirdly on tm updates, so no selecting in tm
+        if (Models.Container.getCurrentContainer() instanceof TradeMarketContainer) return;
+        int slot = hoveredSlot.getContainerSlot();
+
+        // Selecting armor and accessory slots is not allowed
+        if (slot == 9 || slot == 10 || slot == 11 || slot == 12 || slot == 36 || slot == 37 || slot == 38 || slot == 39)
+            return;
 
         ItemStack itemStack = hoveredSlot.getItem();
-        Optional<GearItem> wynnItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
+        if (itemStack.isEmpty()) return;
+
+        Optional<WynnItem> wynnItemOpt = Models.Item.getWynnItem(itemStack);
         if (wynnItemOpt.isEmpty()) return;
+        WynnItem wynnItem = wynnItemOpt.get();
+        if (!(wynnItem instanceof GearItem)) return;
 
-        if (comparedItem == itemStack) {
-            comparedItem = null;
+        if (selectedItems.containsKey(wynnItem)) {
+            selectedItems.remove(wynnItem);
         } else {
-            comparedItem = itemStack;
-            compareToEquipped = false;
+            selectedItems.put(wynnItem, itemStack);
+        }
+    }
+
+    private String getTag(String key) {
+        return I18n.get(key);
+    }
+
+    private List<Component> getWynnOrVanillaLines(
+            AbstractContainerScreen<?> screen, WynnItem wynnItem, ItemStack itemStack) {
+        List<Component> wynnTooltip = TooltipUtils.getWynnItemTooltip(itemStack, wynnItem);
+        return wynnTooltip.isEmpty() ? screen.getTooltipFromItem(McUtils.mc(), itemStack) : wynnTooltip;
+    }
+
+    private MutableComponent getPaddedComponent(String string, int tooltipWidth) {
+        Font font = FontRenderer.getInstance().getFont();
+        // This is not the exact center, but good enough
+        int count = Math.round((tooltipWidth - font.width(string)) / 2.0f / font.width(" "));
+        return Component.literal(" ".repeat(count) + string);
+    }
+
+    private void removeFlavourText(List<Component> lines) {
+        while (lines.getLast().getString().startsWith("§8")) {
+            lines.removeLast();
+        }
+    }
+
+    private void removeSetInfoText(List<Component> lines) {
+        for (int i = lines.size() - 1; i >= 0; --i) {
+            StyledText line = StyledText.fromComponent(lines.get(i)).getNormalized();
+            if (line.getMatcher(WynnItemParser.SET_ITEM_PATTERN).matches()) {
+                lines.remove(i);
+            } else if (line.getMatcher(WynnItemParser.SET_PATTERN).matches()) {
+                lines.remove(i);
+                lines.remove(i - 1);
+                return;
```

Comment:
Why remove just two lines? There are sets with more items in it. Or am I missing something here?

Reply:
> Why remove just two lines? There are sets with more items in it. Or am I missing something here?

First if branch removes item lines eg. "- Morph-Emerald" or "- Morph-Ruby". Second branch is when we hit set name eg. "Morph Set (1/8)", then we remove the set name and an empty line above it.